src/sdg_hub/blocks/llm_utils.py

```
@@ -0,0 +1,166 @@
+            # Initialize all output columns with empty lists
+            matches = {column_name: [] for column_name in self.output_cols}
+
+            # Process tag pairs for matching columns
+            for start_tag, end_tag, output_col in zip(
+                self.start_tags,
+                self.end_tags,
+                self.output_cols,
+            ):
+                matches[output_col] = self._extract_matches(
+                    generated_string, start_tag, end_tag
+                )
```

Pylint (3.3.7)

[refactor] Too many nested blocks (6/5)

(R1702)